### PR TITLE
Refactor folder structure

### PR DESCRIPTION
**Pull request by core23**
I am targeting this branch, because this is BC.

## Changelog

```markdown
### Changed
 - Changed internal folder structure to `src`, `tests` and `docs`
```

## Subject

Adopt the new folder structure discussed on the [dev-kit](https://github.com/sonata-project/dev-kit/issues/278)

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -1,4 +1,4 @@
 .* export-ignore
 *.md export-ignore
 Tests/* export-ignore
-Resources/doc/* export-ignore
+docs/* export-ignore
```
Comment (dbu) [inline .gitattributes]: most packages i have seen use singular "doc" and not docs. i'd prefer that over this, but if this has already been decided i don't mind that much either.

**Comment by OskarStark** [comment]
@core23 can you please check travis?

**Comment by jordisala1991** [comment]
Let's see if we can make it green without rebase

**Comment by jordisala1991** [comment]
nope, you need to rebase @core23

**Comment by jordisala1991** [comment]
Thank you @core23 , at the end we didn't need to rebase :)

**Pull request merged**